core/src/main/java/com/netflix/iceberg/util/BinPacking.java

```diff
@@ -107,7 +113,22 @@ public boolean hasNext() {
           bins.addLast(bin);
 
           if (bins.size() > lookback) {
-            return ImmutableList.copyOf(bins.removeFirst().items());
+            if (expensiveTaskFirst) {
+              Bin maxBin = null;
+
+              // Iterate through all bins looking for one with maximum weight, taking O(n) time.
+              for (Bin currBin : bins) {
```

Comment:
This can probably be replaced by Collections.max(bins, (b1, b2) -> [compare body])

Reply:
I agree. This is a good idea.

Reply:
+1 Done.